src/app/footer/footer.component.ts

```diff
@@ -7,4 +9,16 @@ import { Component } from '@angular/core';
 })
 export class FooterComponent {
 
+  constructor(private route: ActivatedRoute, private userService: UserService, private router: Router) {
+  }
+  routineId: string = this.route.snapshot.params['routine_id'];
+
+  navigateToProfile() {
```

Comment:
Getting rid of this because the html already takes care of navigating users to `/profile` via the `routerLink` attribute